### ISSUE TITLE
Spack is too picky about os compatibility

### ISSUE THREAD

**Issue by CreRecombinase** [spack/spack#15251]
Spack detects a different os depending on whether I'm on a compute node or a login node

on a compute node it's
```
t.cri.nknoblauch@cri16cn084:~:lsb_release -a
LSB Version:    :base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID: RedHatEnterpriseComputeNode
Description:    Red Hat Enterprise Linux ComputeNode release 6.7 (Santiago)
Release:        6.7
Codename:       Santiago
t.cri.nknoblauch@cri16cn084:~:spack arch
linux-redhatenterprisecomputenode6-haswell
```

and on a login node it's:
```
t.cri.nknoblauch@cri16in001:~:lsb_release -a
LSB Version:    :base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID: RedHatEnterpriseServer
Description:    Red Hat Enterprise Linux Server release 6.7 (Santiago)
Release:        6.7
Codename:       Santiago
t.cri.nknoblauch@cri16in001:~:spack arch
linux-rhel6-haswell
```

Unfortunately spack won't let me build across os
```
ValueError: Can't recreate arch for spec linux-rhel6-haswell on current arch linux-redhatenterprisecomputenode6-haswell; spec architecture is too different
```

This wouldn't be a big deal except for the fact that compute nodes don't have access to the internet.

I'm quite sure that `redhatenterprisecomputenode6` and `rhel6` are binary compatible, but I'm not sure how to tell spack that.  The error message `architecture is too different` is kind of tantalizing, because it implies that the architectures don't have to be identical.

**Comment by CreRecombinase** [comment]
I should add that the other frustrating thing about this is that if I bite the bullet and install 2 versions of every package, `spack load foo` gives me an error
```
t.cri.nknoblauch@cri16in001:~:spack load python
==> Error: python matches multiple packages.
  Matching packages:
    x2emmmu python@3.7.6%gcc@9.2.0 arch=linux-redhatenterprisecomputenode6-haswell
    t5f6yk4 python@3.7.6%gcc@9.2.0 arch=linux-rhel6-haswell
```

**Comment by adamjstewart** [comment]
Closed by #15253
